client.go

```diff
@@ -428,6 +446,38 @@ func (c *Client) handleMessage(msgType int, data []byte) error {
 				return err
 			}
 			c.sigChainBlockHash = sigChainBlockHash
+
+		// client auth
+		case authChallengeAction:
+			var challenge string
+			if err := json.Unmarshal(*msg["Challenge"], &challenge); err != nil {
+				return err
+			}
+			byteChallenge, err := hex.DecodeString(challenge)
+			if err != nil {
+				return err
+			}
+
+			clientSalt := make([]byte, 32)
```

Comment:
There is an existing helper function `RandomBytes()`

Reply:
Yes, modified.